backend/src/controllers/studentsControllers.js

```diff
@@ -0,0 +1,136 @@
+import { createStudent } from "../repository/createStudent.js";
+import { createUser } from "../repository/createUser.js";
+import { deleteUser as deleteUser } from "../repository/deleteStudent.js";
+import { getStudent } from "../repository/getStudent.js";
+import { getStudents } from "../repository/getStudents.js";
+import { updateStudent } from "../repository/updateStudent.js";
+
+export const getStudentsController = async (req, res) => {
+  try {
+    const { name, surname, facultyId, year, academicGroupId } = req.query;
+
+    const students = await getStudents(
+      name,
+      surname,
+      facultyId,
+      year,
+      academicGroupId
+    );
+
+    res.status(200).json(students);
+  } catch (err) {
+    res.status(500).json({ error: "Failed to fetch students" });
+  }
+};
+
+export const getStudentByUsernameController = async (req, res) => {
+  try {
+    const { username } = req.params;
+
+    const student = await getStudent(username);
+    const studentFound = student.length === 1;
+
+    res.status(studentFound ? 200 : 404).json(student);
+  } catch (err) {
+    res.status(500).json({ error: "Failed to fetch students" });
```

Comment:
Čia visur error grąžini tą patį "Failed to fetch students". Tai čia sunku bus žinot kuris kontroleris čia failina. Tai gal tą kaip sakiau geriau daryt, throwini errorą iš repo, čia catchini, ir pasini error: err.error